server.go

```diff
@@ -3126,3 +3126,50 @@ func (s *server) watchChannelStatus() {
 		}
 	}
 }
+
+// isNodePublic attempts to determine if a node with the given public key is
+// publicly advertised within the network.
+func (s *server) isNodePublic(pubKey *btcec.PublicKey) (bool, error) {
```

Comment:
Seems like we can pull this into a method directly on the node itself in the database. This would allow us to unit test this section with a small mock graph, possibly writing some helper methods to make it easier to make table-driven tests for this use case.

Reply:
Fixed.